pkg/doc/signature/jsonld/processor.go

```diff
@@ -506,34 +507,43 @@ func (p *Processor) normalizeFilteredDataset(view string) (string, error) {
 	return result.(string), nil
 }
 
-func useDocumentLoader(ldOptions *ld.JsonLdOptions, loader ld.DocumentLoader, cache map[string]interface{}) {
-	if loader == nil && len(cache) == 0 {
+func useDocumentLoader(ldOptions *ld.JsonLdOptions, loader ld.DocumentLoader, cache *sync.Map) {
+	if loader == nil && cache == nil {
 		return
 	}
 
 	ldOptions.DocumentLoader = getCachingDocumentLoader(loader, cache)
 }
 
-func getCachingDocumentLoader(loader ld.DocumentLoader, cache map[string]interface{}) *ld.CachingDocumentLoader {
+func getCachingDocumentLoader(loader ld.DocumentLoader, cache *sync.Map) *jsonld.CachingDocumentLoader {
 	cachingLoader := createCachingDocumentLoader(loader)
 
-	for k, v := range cache {
-		cachingLoader.AddDocument(k, v)
+	if cache != nil {
+		cache.Range(func(key, value interface{}) bool {
+			k, ok := key.(string)
+			if !ok {
+				return false
+			}
+
+			cachingLoader.AddDocument(k, value)
```

Comment:
wouldn't this add the same documents again and again to `cachingLoader`?

Reply:
it's adding from cache into cacheLoader, just like the original logic, it's just the way looping through cache is different because of its `*sync.Map` type